homeassistant/components/zha/__init__.py

```diff
@@ -359,6 +372,90 @@ def zdo_command(self, tsn, command_id, args):
         pass
 
 
+class ZhaEndpointEntity(entity.Entity):
+    """A base class for ZHA endpoints."""
+
+    _domain = DOMAIN
+
+    def __init__(self, endpoint, manufacturer, model, application_listener,
+                 unique_id, keepalive_interval=7200, **kwargs):
+        """Init ZHA endpoint entity."""
+        self._device_state_attributes = {}
+        ieee = endpoint.device.ieee
+        ieeetail = ''.join(['%02x' % (o, ) for o in ieee[-4:]])
+        if manufacturer and model is not None:
+            self.entity_id = "{}.{}_{}_{}_{}".format(
+                self._domain,
+                slugify(manufacturer),
+                slugify(model),
+                ieeetail,
+                endpoint.endpoint_id,
+            )
+            self._device_state_attributes['friendly_name'] = "{} {}".format(
+                manufacturer,
+                model,
+            )
+        else:
+            self.entity_id = "{}.zha_{}_{}".format(
+                self._domain,
+                ieeetail,
+                endpoint.endpoint_id,
+            )
+
+        nwk = endpoint.device.nwk
+        self._device_state_attributes['nwk'] = '0x{0:04x}'.format(nwk)
+        self._device_state_attributes['ieee'] = str(endpoint.device.ieee)
+        self._device_state_attributes['last_update'] = None
+        self._device_state_attributes['lqi'] = endpoint.device.lqi
+        self._device_state_attributes['rssi'] = endpoint.device.rssi
+
+        self._endpoint = endpoint
+        self._state = 'offline'
+        self._unique_id = unique_id
+        self._keepalive_interval = keepalive_interval
+
+        application_listener.register_entity(ieee, self)
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._unique_id
+
+    @property
+    def state(self) -> str:
+        """Return the state of the entity."""
+        return self._state
+
+    @property
+    def hidden(self) -> bool:
+        """Hide by default."""
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        update_time = None
+        if self._endpoint.device.last_seen is not None:
+            time_struct = time.localtime(self._endpoint.device.last_seen)
+            update_time = time.strftime("%Y-%m-%dT%H:%M:%S", time_struct)
+        self._device_state_attributes['last_update'] = update_time
```

Comment:
We should not store update time.

Reply:
Maybe update_time or last_update is not the best name. This is essentially the time when the last message was received from the device. I thought this might be useful for verification or troubleshooting purposes.